### PR TITLE
Clarify errors when plugins fail to load

### PR DESCRIPTION
**Pull request by mjcarroll**
# 🦟 Bug fix

## Summary

I find that it's hard to tell one plugin failure mode from another with the current error messages.  This expands the messages and also includes relevant information for helping to diagnose system plugin issues.

Now in the case that it fails to find a library, gz-sim will output the list of paths that were searched for the library.

When a bad (or misspelled!) plugin name is provided, it will output a list of the known plugins and aliases in that shared library.

Example Improved output:
```
[ RUN      ] SystemLoader.BadLibraryPath
[Err] [SystemLoader.cc:102] Failed to load system plugin: (Reason: Could not find shared library)
- Requested plugin name: [gz::sim::systems::Physics]
- Requested library name: [foo.so]
- Library search paths:
  - /home/mjcarroll/workspaces/gz_garden/build/gz-sim7/lib/
  - /home/mjcarroll/.gz/sim/plugins/
  - /home/mjcarroll/.ignition/gazebo/plugins/
  - /home/mjcarroll/workspaces/gz_garden/install/lib/gz-sim-7/plugins/
[       OK ] SystemLoader.BadLibraryPath (79 ms)
[ RUN      ] SystemLoader.BadPluginName
[Err] [SystemLoader.cc:164] Failed to load system plugin: (Reason: library does not contain requested plugin)
- Requested plugin name: [gz::sim::systems::Foo]
- Requested library name: [libgz-sim7-physics-system.so]
- Resolved library path: [/home/mjcarroll/workspaces/gz_garden/build/gz-sim7/lib/libgz-sim7-physics-system.so]
- Detected Plugins:
  - gz::sim::v7::systems::Physics
    aliases:
      gz::sim::systems::Physics
      ignition::gazebo::systems::Physics
[       OK ] SystemLoader.BadPluginName (43 ms)

```

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
linters

File: src/SystemLoader.cc [hunk 1]
```diff
@@ -101,36 +115,83 @@ class gz::sim::SystemLoaderPrivate
...
+      std::stringstream ss;
+      ss << "Failed to load system plugin: "
+         << "(Reason: library does not contain requested plugin)\n"
+         << "- Requested plugin name: [" << _sdfPlugin.Name() << "]\n"
+         << "- Requested library name: [" << _sdfPlugin.Filename() << "]\n"
+         << "- Resolved library path: [" << pathToLib << "]\n"
+         << "- Detected Plugins:\n";
+      for (const auto &pluginIt : pluginNames)
+      {
+        ss << "  - " << pluginIt;
```
Comment (JoanAguilar) [inline src/SystemLoader.cc:153]:
```suggestion
        ss << "  - " << pluginIt << "\n";
```

**Review by JoanAguilar — APPROVED** [review]
I don't think I can't say much of the unit tests, the rest looks good to me (besides a minor comment).

**Comment by mjcarroll** [comment]
> I don't think I can't say much of the unit tests, the rest looks good to me (besides a minor comment).

I updated the tests to actually check the output for the general failure modes.

File: src/SystemLoader_TEST.cc [hunk 2]
```diff
@@ -15,6 +15,7 @@
  *
 */
 
+#include "gtest/gtest.h"
```
Comment (ahcorde) [inline src/SystemLoader_TEST.cc:18]: remove?

File: src/SystemLoader.cc [hunk 3]
```diff
@@ -101,36 +115,83 @@ class gz::sim::SystemLoaderPrivate
     auto pluginNames = this->loader.LoadLib(pathToLib, true);
     if (pluginNames.empty())
     {
-      gzerr << "Failed to load system plugin [" << filename <<
-                "] : couldn't load library on path [" << pathToLib <<
-                "]." << std::endl;
+      std::stringstream ss;
```
Comment (ahcorde) [inline src/SystemLoader.cc:118]:
```cpp
#include <sstream>
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Pull request merged**